packages/lodestar/src/chain/validation/aggregateAndProof.ts

```diff
@@ -1,143 +1,122 @@
-  } catch (error) {
-    throw new AttestationError({
-      code: AttestationErrorCode.AGGREGATOR_NOT_IN_COMMITTEE,
-      job: attestationJob,
-    });
+  // [REJECT] The aggregator's validator index is within the committee
+  // -- i.e. aggregate_and_proof.aggregator_index in get_beacon_committee(state, aggregate.data.slot, aggregate.data.index).
+  if (!committeeIndices.includes(aggregateAndProof.aggregatorIndex)) {
+    throw new AttestationError({code: AttestationErrorCode.AGGREGATOR_NOT_IN_COMMITTEE});
   }
 
-  if (!committee.includes(aggregateAndProof.message.aggregatorIndex)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.AGGREGATOR_NOT_IN_COMMITTEE,
-      job: attestationJob,
-    });
-  }
-
-  if (!isAggregatorFromCommitteeLength(committee.length, aggregateAndProof.message.selectionProof)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.INVALID_AGGREGATOR,
-      job: attestationJob,
-    });
-  }
-
-  const slot = attestation.data.slot;
-  const epoch = computeEpochAtSlot(slot);
-  const indexedAttestation = attestationTargetState.getIndexedAttestation(attestation);
-  const aggregator = attestationTargetState.index2pubkey[aggregateAndProof.message.aggregatorIndex];
-
+  // [REJECT] The aggregate_and_proof.selection_proof is a valid signature of the aggregate.data.slot
+  // by the validator with index aggregate_and_proof.aggregator_index.
+  // [REJECT] The aggregator signature, signed_aggregate_and_proof.signature, is valid.
+  // [REJECT] The signature of aggregate is valid.
+  const aggregator = targetState.index2pubkey[aggregateAndProof.aggregatorIndex];
   const signatureSets = [
-    getSelectionProofSignatureSet(config, attestationTargetState, slot, aggregator, aggregateAndProof),
-    getAggregateAndProofSignatureSet(config, attestationTargetState, epoch, aggregator, aggregateAndProof),
-    allForks.getIndexedAttestationSignatureSet(attestationTargetState, indexedAttestation),
+    getSelectionProofSignatureSet(targetState, attSlot, aggregator, signedAggregateAndProof),
+    getAggregateAndProofSignatureSet(targetState, attEpoch, aggregator, signedAggregateAndProof),
+    allForks.getIndexedAttestationSignatureSet(targetState, indexedAttestation),
   ];
-
   if (!(await chain.bls.verifySignatureSets(signatureSets))) {
-    throw new AttestationError({
-      code: AttestationErrorCode.INVALID_SIGNATURE,
-      job: attestationJob,
-    });
+    throw new AttestationError({code: AttestationErrorCode.INVALID_SIGNATURE});
   }
 
-  // TODO: once we have pool, check if aggregate block is seen and has target as ancestor
-
-  // verifySignature = false, verified in batch above
-  if (!phase0.isValidIndexedAttestation(attestationTargetState, indexedAttestation, false)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.INVALID_INDEXED_ATTESTATION,
-      job: attestationJob,
-    });
+  if (chain.seenAggregators.isKnown(targetEpoch, aggregatorIndex)) {
```

Comment:
this is a lightweight check, should we move it to the early stage?

Reply:
this is a duplicated, checked above

Reply:
The duplication is intentional, since there as async call before to the bls verification there's a race condition.

I should add a comment there tho